framework/src/main/java/org/checkerframework/framework/type/SupertypeFinder.java

```diff
@@ -216,28 +216,50 @@
      */
     private Map<TypeVariable, AnnotatedTypeMirror> getTypeVarToTypeArg(AnnotatedDeclaredType type) {
       Map<TypeVariable, AnnotatedTypeMirror> mapping = new HashMap<>();
-      AnnotatedDeclaredType enclosing = type;
-      while (enclosing != null) {
-        TypeElement enclosingTypeElement = (TypeElement) enclosing.getUnderlyingType().asElement();
-        List<? extends TypeParameterElement> typeParams = enclosingTypeElement.getTypeParameters();
-        List<AnnotatedTypeMirror> typeArgs = enclosing.getTypeArguments();
-        for (int i = 0; i < enclosing.getTypeArguments().size(); ++i) {
-          AnnotatedTypeMirror typArg = typeArgs.get(i);
-          TypeParameterElement ele = typeParams.get(i);
-          mapping.put((TypeVariable) ele.asType(), typArg);
-        }
+      // addTypeVarsFromEnclosingTypes can't be called with `type` because it calls
+      // `directSupertypes(types)`, which then calls this method. Add the type variables from `type`
+      // and then recur on the enclosing type and super types.
```

Comment:
Can you clarify where the recursion is?  There used to be a call to `getTypeVarToTypeArg` below, but this diff [removes it](https://github.com/typetools/checker-framework/pull/4931/files#diff-e10288bc0474b1292d46708ba01d05eb8501722ec6737da6b9a3e50f0c9c4960L234).  `addTypeVarsFromEnclosingTypes` is recursive, but I wouldn't expect a comment about that fact here in this method.